arches/app/models/models.py

```diff
@@ -1280,6 +1252,118 @@ class Meta:
         permissions = (("no_access_to_resourceinstance", "No Access"),)
 
 
+class ResourceInstanceLifecycle(models.Model):
+    id = models.UUIDField(primary_key=True, serialize=False)
+    name = models.TextField()
+
+    def get_initial_resource_instance_lifecycle_state(self):
+        return self.resource_instance_lifecycle_states.get(is_initial_state=True)
+
+    def serialize(self, fields=None, exclude=None, **kwargs):
+        ret = JSONSerializer().handle_model(
+            self, fields=fields, exclude=exclude, **kwargs
+        )
+
+        ret["resource_instance_lifecycle_states"] = [
+            JSONSerializer().handle_model(resource_instance_lifecycle_state)
+            for resource_instance_lifecycle_state in self.resource_instance_lifecycle_states.all()
+        ]
+
+        return ret
+
+    class Meta:
+        db_table = "resource_instance_lifecycles"
+        managed = True
+
+
+class ResourceInstanceLifecycleState(models.Model):
+    id = models.UUIDField(primary_key=True, serialize=False)
+    name = I18n_TextField()
+    action_label = I18n_TextField()
+    is_initial_state = models.BooleanField(default=False)
+    can_delete_resource_instances = models.BooleanField(default=False)
+    can_edit_resource_instances = models.BooleanField(default=False)
+    resource_instance_lifecycle = models.ForeignKey(
+        on_delete=models.CASCADE,
+        to="models.ResourceInstanceLifecycle",
+        related_name="resource_instance_lifecycle_states",
+    )
+    previous_resource_instance_lifecycle_states = models.ManyToManyField(
+        "self",
+        through="ResourceInstanceLifecycleStateFromXRef",
+        symmetrical=False,
+        related_name="next_lifecycle_states",
+    )
+    next_resource_instance_lifecycle_states = models.ManyToManyField(
+        "self",
+        through="ResourceInstanceLifecycleStateToXRef",
+        symmetrical=False,
+        related_name="previous_lifecycle_states",
+    )
+
+    def serialize(self, fields=None, exclude=None, **kwargs):
+        ret = JSONSerializer().handle_model(
```

Comment:
Can you flesh this out a bit more? Just want to be sure I'm understanding fully.

Reply:
Sure, my experience was that when serializing, only the active language's value got serialized, which is a problem if that is then deserialized somewhere else (accepted from an AJAX request, or a roundtrip we're doing ourselves in [resource_graphs/exporter.py](https://github.com/archesproject/arches/pull/11096/files#diff-4ef90fb853af3f36fd77af185226287e27a047f38c12c781dcb513f655d658eb).

I guess at the time I wrote this comment I thought I saw an example of json being used to instantiate a python model directly, but maybe not. Finding the exact invocation where the data is being deserialized is not really the point, the point is just that we probably shouldn't be satisfied with only serializing the value in the active language, as it will probably lead to data loss.

Maybe we bounce this to a new ticket as it has more to do with our serializers.

Reply:
Yep, sounds like an issue with the serializer instead 👍

Reply:
Yep, sounds like an issue with the serializer instead 👍